Support for DataTypes/CanSee

ttung opened the pull request:
### Description
This adds the part of the schema for DataTypes and CanSee.

Depends on #30, #40, #46, #47, #48, #50, #51, #52, #55

#### Issue
[ch64743](https://app.clubhouse.io/genepi/stories/space/64743)

### Test plan
1. Ran tests!
2. Created a db from scratch and observed that the "enum"s were created.
3. Created a db from the prior revision, applied the autogenerated script, and observed that the "enum"s were created.

ttung commented:
This basically works but python packaging is thwarting me!

phoenixAja commented on src/py/aspen/database/models.py:
```diff
@@ -27,14 +35,32 @@
+
+# Mypy gets confused about whether sqlalchemy enum columns are strings or enums, see here:
+# https://github.com/dropbox/sqlalchemy-stubs/issues/114
+# This is the (gross) workaround. Keep an eye on the issue and get rid of it once it's fixed.
```
lol

phoenixAja commented on src/py/aspen/database/models.py:
```diff
@@ -65,11 +91,70 @@ def __repr__(self):
+)
+
+
+def _after_datatypes_create(target, connection, **kw):
+    for datatype in DataType:
+        connection.execute(
+            text(
+                f"INSERT INTO {target.schema}.{target.name} (item_id) VALUES (:name)"
+            ).params(name=datatype.value)
+        )
+
+
+event.listen(_DataTypeTable.__table__, "after_create", _after_datatypes_create)
```
i'm a little confused on how this is going to be used

ttung replied:
For enum tables, we need to pre-create the enum types in the table.  This is a hook that allows some code to be called after the table is created, and we shove in the enum types into the table at that time.

phoenixAja replied:
so we'll need to create similar listeners for any enum we create in the future?

ttung replied:
Yes, that's correct.

Actually, you gave me a great idea.  We should fold that logic into the `EnumTable` function.

phoenixAja replied:
yeah that's actually why i was a little confused, i was thinking that we should fold that logic in there instead. No rush though!

ttung replied:
Done in #61

phoenixAja commented on .hgtags:
```diff
@@ -0,0 +1,5 @@
+624b2a967271f0cc24b0f6af19ae73fc4bdbd634 t1
+3bf4e2032f47137600539a08db24b9341c0dc838 t1
+624b2a967271f0cc24b0f6af19ae73fc4bdbd634 t1
+624b2a967271f0cc24b0f6af19ae73fc4bdbd634 t1
+c32aa987429ba54e4c7be9dd679a8a9c76fde661 t1
```
what is this file for?

ttung replied:
It should be gone now.

Commit:
Support for DataTypes/CanSee

### Description
This adds the part of the schema for DataTypes and CanSee.

#### Issue
[ch64743](https://app.clubhouse.io/genepi/stories/space/64743)

### Test plan
1. Ran tests!
2. Created a db from scratch and observed that the "enum"s were created.
3. Created a db from the prior revision, applied the autogenerated script, and observed that the "enum"s were created.
